### DISCUSSION TITLE
TabMixPlus Reopen Last Closed Tabs button not working anymore after Nightly updated today

### DISCUSSION THREAD

**Discussion by Shahius** [onemen/TabMixPlus#223] [Q&A] [answered]
Hello, dear onemen,

TabMixPlus Reopen Last Closed Tabs  green button is not working anymore after Nightly's update. It happened on both my computers today. Before the update everything worked perfectly. 
Pressing this green button (or the links from this opening closed tabs list) also does nothing. 
I have the latest 1.0.0-pre.15 build installed.

I took a couple of screenshots of this bug:

https://imgur.com/a/IaGP8xN

Thanks.

P.S. I would like to add a screenshot of Browser Console:
https://imgur.com/a/TV916r9

I am not a real tek-savvy, so not sure if it helps.

**Comment by onemen** [comment]
Firefox bug [1819675](https://bugzilla.mozilla.org/show_bug.cgi?id=1819675) - rename SessionStore.getClosedTabCount and getClosedTabData

there was some other change lately

it will work on this next week

**Comment by onemen** [comment]
@Shahius 

download and install this [test build](https://tabmixplus.org/dev-builds/test%20builds/1.0.0-pre.15.20230429.1.xpi)

**Reply by Shahius** [reply]
Just did it. Looks like nothing changed for me...

**Reply by Shahius** [reply]
Applying chrome://tabmixplus/content/tabmix.xhtml to chrome://browser/content/browser.xhtml Overlays.jsm:92:15
Applying chrome://tabmixplus/content/overlay/tabContextMenu.xhtml to chrome://browser/content/browser.xhtml Overlays.jsm:92:15
Loading eval'd script into chrome://browser/content/browser.xhtml Overlays.jsm:546:15
Tabmix initializer: 
Object { key: "init", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "beforeStartup", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "onContentLoaded", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "beforeBrowserInitOnLoad", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "onWindowOpen", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "afterDelayedStartup", phase: {…} }
tabmix.js:1174:17
The resource at “https://a.pikabu.ru/evl.php” was blocked by OpaqueResponseBlocking. Reason: “Javascript validation failed”. 8
Security Error: Content at chrome://browser/skin/tabs.css may not load or link to moz-extension://2ddfff31-1214-47bd-b899-6c91ff2224a8/background.html. 2
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    ct_setButtonDisableState chrome://tabmixplus/content/session/sessionStore.js:355
    observe chrome://tabmixplus/content/session/sessionStore.js:826
    _notifyOfClosedObjectsChange resource:///modules/sessionstore/SessionStore.sys.mjs:5322
    notify resource://gre/modules/Timer.sys.mjs:49
sessionStore.js:363:40
AbortError: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved ConduitsParent.sys.mjs:362
Security Error: Content at chrome://browser/skin/tabs.css may not load or link to moz-extension://2ddfff31-1214-47bd-b899-6c91ff2224a8/background.html. 4
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    ct_setButtonDisableState chrome://tabmixplus/content/session/sessionStore.js:355
    observe chrome://tabmixplus/content/session/sessionStore.js:826
    _notifyOfClosedObjectsChange resource:///modules/sessionstore/SessionStore.sys.mjs:5322
    notify resource://gre/modules/Timer.sys.mjs:49
sessionStore.js:363:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabData is not a function
    get getClosedTabData chrome://tabmixplus/content/session/sessionStore.js:370
    populateUndoSubmenu chrome://tabmixplus/content/session/sessionStore.js:392
    populateUndoSubmenu chrome://tabmixplus/content/changecode.js line 222 > eval:28
    populateClosedTabsMenu chrome://tabmixplus/content/session/sessionStore.js:872
    initObjectPanel chrome://tabmixplus/content/session/sessionStore.js:785
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:189
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:1483
    _blockersPromise resource:///modules/PanelMultiView.sys.mjs:223
sessionStore.js:370:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    _undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:650
    ct_undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:735
    ct_window_undoCloseTab chrome://tabmixplus/content/tabmix.js:34
    oncommand chrome://browser/content/browser.xhtml:1
sessionStore.js:363:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    _undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:650
    ct_undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:735
    handleButtonEvent chrome://tabmixplus/content/session/sessionStore.js:485
    oncommand chrome://browser/content/browser.xhtml:1
sessionStore.js:363:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabData is not a function
    get getClosedTabData chrome://tabmixplus/content/session/sessionStore.js:370
    populateUndoSubmenu chrome://tabmixplus/content/session/sessionStore.js:392
    populateUndoSubmenu chrome://tabmixplus/content/changecode.js line 222 > eval:28
    populateClosedTabsMenu chrome://tabmixplus/content/session/sessionStore.js:872
    initObjectPanel chrome://tabmixplus/content/session/sessionStore.js:785
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:189
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:1483
    _blockersPromise resource:///modules/PanelMultiView.sys.mjs:223
sessionStore.js:370:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    _undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:650
    ct_undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:735
    ct_window_undoCloseTab chrome://tabmixplus/content/tabmix.js:34
    oncommand chrome://browser/content/browser.xhtml:1
sessionStore.js:363:40
Security Error: Content at chrome://browser/skin/tabs.css may not load or link to moz-extension://2ddfff31-1214-47bd-b899-6c91ff2224a8/background.html.
Referrer Policy: Ignoring the less restricted referrer policy “no-referrer-when-downgrade” for the cross-site request: https://raw.githubusercontent.com/onemen/TabMixPlus/main/addon/icon.png icon.png
can't access property "isAdoptingTab", this.window.gBrowserInit is undefined ext-browser.js:1134
Uncaught NS_ERROR_FILE_NOT_FOUND: 
    get _pathRegExp chrome://tabmix-resource/content/log.jsm:236
    _formatStack chrome://tabmix-resource/content/log.jsm:241
    TMP_console_assert chrome://tabmix-resource/content/log.jsm:310
    TMP_console_wrapper chrome://tabmixplus/content/utils.js:260
    TMP_console_wrapper chrome://tabmixplus/content/utils.js:292
    TMP_updateMainContextMenu chrome://tabmixplus/content/click/click.js:855
    handleEvent chrome://tabmixplus/content/click/click.js:482
    openContextMenu chrome://browser/content/nsContextMenu.js:106
    receiveMessage resource:///actors/ContextMenuParent.sys.mjs:28
log.jsm:236

**Reply by Shahius** [reply]
Again, sorry if this log is not what  you need.

**Comment by onemen** [comment]
delete Tabmix
close Firefox
open Firefox
download new test build
go to about:support 'clear startup cache'
install Tabmix from the file you downloaded

**Reply by Shahius** [reply]
Just did exactly what you said, step by step. No luck.

1685892121823	addons.xpi	WARN	Checking C:\Program Files\Firefox Nightly\distribution\extensions for addons
Error: Cannot find loader for bootstrap XPIProvider.jsm:1935:15
1685892122096	addons.webextension.bootstrap@xxf	WARN	Loading extension 'bootstrap@xxf': Reading manifest: Warning processing version: version must be a version string consisting of at most 4 integers of at most 9 digits without leading zeros, and separated with dots
1685892122186	addons.webextension.gmailnoads@mywebber.com	WARN	Loading extension 'gmailnoads@mywebber.com': Reading manifest: Warning processing options_page: An unexpected property was found in the WebExtension manifest.
1685892122197	addons.webextension.jid1-MnnxcxisBPnSXQ@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ@jetpack': Reading manifest: Warning processing storage: An unexpected property was found in the WebExtension manifest.
No chrome package registered for chrome://userchromejs/content/BootstrapLoader.jsm
Applying chrome://tabmixplus/content/tabmix.xhtml to chrome://browser/content/browser.xhtml Overlays.jsm:92:15
Applying chrome://tabmixplus/content/overlay/tabContextMenu.xhtml to chrome://browser/content/browser.xhtml Overlays.jsm:92:15
Loading eval'd script into chrome://browser/content/browser.xhtml Overlays.jsm:546:15
Tabmix initializer: 
Object { key: "init", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "beforeStartup", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "onContentLoaded", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "beforeBrowserInitOnLoad", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "onWindowOpen", phase: {…} }
tabmix.js:1174:17
Tabmix initializer: 
Object { key: "afterDelayedStartup", phase: {…} }
tabmix.js:1174:17
Uncaught (in promise) NS_ERROR_FILE_NOT_FOUND: 
    reportError chrome://tabmix-resource/content/log.jsm:339
    TMP_console_wrapper chrome://tabmixplus/content/utils.js:260
    TMP_console_wrapper chrome://tabmixplus/content/utils.js:292
log.jsm:339
In add-on jid0-GjwrPchS3Ugt7xydvqVK4DQk8Ls@jetpack, attempting to use listener "notifications.onButtonClicked", which is unimplemented. ExtensionParent.sys.mjs:1254:13
can't access property "spec", browser.contentPrincipal is undefined 8 browser-fullZoom.js:734
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    ct_setButtonDisableState chrome://tabmixplus/content/session/sessionStore.js:355
    observe chrome://tabmixplus/content/session/sessionStore.js:826
    _notifyOfClosedObjectsChange resource:///modules/sessionstore/SessionStore.sys.mjs:5322
    notify resource://gre/modules/Timer.sys.mjs:49
sessionStore.js:363:40
In add-on {d10d0bf8-f5b5-c8b4-a8b2-2b9879e08c5d}, attempting to use listener "notifications.onButtonClicked", which is unimplemented. ExtensionParent.sys.mjs:1254:13
Uncaught TypeError: TabmixSvc.ss.getClosedTabData is not a function
    get getClosedTabData chrome://tabmixplus/content/session/sessionStore.js:370
    populateUndoSubmenu chrome://tabmixplus/content/session/sessionStore.js:392
    populateUndoSubmenu chrome://tabmixplus/content/changecode.js line 222 > eval:28
    populateClosedTabsMenu chrome://tabmixplus/content/session/sessionStore.js:872
    initObjectPanel chrome://tabmixplus/content/session/sessionStore.js:785
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:189
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:1483
    _blockersPromise resource:///modules/PanelMultiView.sys.mjs:223
sessionStore.js:370:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    _undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:650
    ct_undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:735
    ct_window_undoCloseTab chrome://tabmixplus/content/tabmix.js:34
    oncommand chrome://browser/content/browser.xhtml:1
sessionStore.js:363:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    _undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:650
    ct_undoCloseTab chrome://tabmixplus/content/session/sessionStore.js:735
    handleButtonEvent chrome://tabmixplus/content/session/sessionStore.js:485
    oncommand chrome://browser/content/browser.xhtml:1
2 sessionStore.js:363:40
Uncaught TypeError: TabmixSvc.ss.getClosedTabData is not a function
    get getClosedTabData chrome://tabmixplus/content/session/sessionStore.js:370
    populateUndoSubmenu chrome://tabmixplus/content/session/sessionStore.js:392
    populateUndoSubmenu chrome://tabmixplus/content/changecode.js line 222 > eval:28
    populateClosedTabsMenu chrome://tabmixplus/content/session/sessionStore.js:872
    initObjectPanel chrome://tabmixplus/content/session/sessionStore.js:785
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:189
    dispatchCustomEvent resource:///modules/PanelMultiView.sys.mjs:1483
    _blockersPromise resource:///modules/PanelMultiView.sys.mjs:223
sessionStore.js:370:40
can't access property "isAdoptingTab", this.window.gBrowserInit is undefined ext-browser.js:1134
Security Error: Content at chrome://browser/skin/tabs.css may not load or link to moz-extension://2ddfff31-1214-47bd-b899-6c91ff2224a8/background.html.
Key event not available on GTK2: key=“u” modifiers=“accel,shift” id=“” browser.xhtml
Key event not available on some keyboard layouts: key=“i” modifiers=“accel,alt,shift” id=“key_browserToolbox” browser.xhtml
Key event not available on GTK2: key=“u” modifiers=“accel,shift” id=“key_tm_mergeWin” browser.xhtml
NS_ERROR_UNEXPECTED: Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIPrefBranch.getBoolPref] 2 NetworkGeolocationProvider.sys.mjs:354
Security Error: Content at chrome://browser/skin/tabs.css may not load or link to moz-extension://2ddfff31-1214-47bd-b899-6c91ff2224a8/background.html.
Uncaught TypeError: TabmixSvc.ss.getClosedTabCount is not a function
    get count chrome://tabmixplus/content/session/sessionStore.js:363
    ct_setButtonDisableState chrome://tabmixplus/content/session/sessionStore.js:355
    observe chrome://tabmixplus/content/session/sessionStore.js:826
    _notifyOfClosedObjectsChange resource:///modules/sessionstore/SessionStore.sys.mjs:5322
    notify resource://gre/modules/Timer.sys.mjs:49
sessionStore.js:363:40


I suppose, I will have to refresh my browser and disable all my extensions to test it correctly. I will do it later today.

**Comment by onemen** [comment, marked as answer]
my bad, i've linked the wrong file
https://tabmixplus.org/dev-builds/test%20builds/1.0.0-pre.16.20230603.1.xpi

**Reply by Shahius** [reply]
Wow! Working now!!!
